libraries/push/impl/src/main/kotlin/io/element/android/libraries/push/impl/notifications/factories/NotificationFactory.kt

```diff
@@ -0,0 +1,295 @@
+            .build()
+    }
+
+    /**
+     * Create the summary notification.
+     */
+    fun createSummaryListNotification(
+        sessionId: SessionId,
+        style: NotificationCompat.InboxStyle?,
+        compatSummary: String,
+        noisy: Boolean,
+        lastMessageTimestamp: Long
+    ): Notification {
+        val accentColor = ContextCompat.getColor(context, R.color.notification_accent_color)
+        val smallIcon = R.drawable.ic_notification
+        val channelId = notificationChannels.getChannelIdForMessage(noisy)
+        return NotificationCompat.Builder(context, channelId)
+            .setOnlyAlertOnce(true)
+            // used in compat < N, after summary is built based on child notifications
+            .setWhen(lastMessageTimestamp)
+            .setStyle(style)
+            .setContentTitle(sessionId.value)
+            .setCategory(NotificationCompat.CATEGORY_MESSAGE)
+            .setSmallIcon(smallIcon)
+            // set content text to support devices running API level < 24
+            .setContentText(compatSummary)
+            .setGroup(sessionId.value)
+            // set this notification as the summary for the group
+            .setGroupSummary(true)
+            .setColor(accentColor)
+            .apply {
+                if (noisy) {
+                    // Compat
+                    priority = NotificationCompat.PRIORITY_DEFAULT
+                    /*
+                    vectorPreferences.getNotificationRingTone()?.let {
+                        setSound(it)
+                    }
+                     */
+                    setLights(accentColor, 500, 500)
+                } else {
+                    // compat
+                    priority = NotificationCompat.PRIORITY_LOW
+                }
+            }
+            .setContentIntent(pendingIntentFactory.createOpenSessionPendingIntent(sessionId))
+            .setDeleteIntent(pendingIntentFactory.createDismissSummaryPendingIntent(sessionId))
+            .build()
+    }
+
+    fun createDiagnosticNotification(): Notification {
+        return NotificationCompat.Builder(context, notificationChannels.getChannelIdForTest())
+            .setContentTitle(buildMeta.applicationName)
+            .setContentText(stringProvider.getString(R.string.notification_test_push_notification_content))
+            .setSmallIcon(R.drawable.ic_notification)
+            .setLargeIcon(getBitmap(R.drawable.element_logo_green))
+            .setColor(ContextCompat.getColor(context, R.color.notification_accent_color))
+            .setPriority(NotificationCompat.PRIORITY_MAX)
+            .setCategory(NotificationCompat.CATEGORY_STATUS)
+            .setAutoCancel(true)
+            .setContentIntent(pendingIntentFactory.createTestPendingIntent())
+            .build()
+    }
+
+    private fun getBitmap(@DrawableRes drawableRes: Int): Bitmap? {
+        val drawable = ResourcesCompat.getDrawable(context.resources, drawableRes, null) ?: return null
```

Comment:
I'm not sure if we'll ever need to use it, but maybe the `theme` parameter set as null here could be passed in the function call, but default to null?